src/diffusers/models/embeddings.py

```diff
@@ -371,6 +414,18 @@ def forward(self, text_embeds: torch.Tensor, image_embeds: torch.Tensor):
         embeds = torch.cat(
             [text_embeds, image_embeds], dim=1
         ).contiguous()  # [batch, seq_length + num_frames x height x width, channels]
+
+        if self.use_positional_embeddings:
+            if embeds.size(1) > self.pos_embedding.size(1):
+                raise ValueError(
+                    "You are trying to generate at a resolution, or higher number of frames, or longer maximum prompt "
+                    "sequence length than what is supported in the `CogVideoXTransformer3D` model. In order to generate "
+                    "at the resolution/num_frames you desire, configure the transformer initialization attributes "
+                    "`sample_height`, `sample_width`, `sample_frames` and `max_text_seq_length` appropriately when "
+                    "initializing your model either with `.from_pretrained(...)` or `.from_config(...)`"
+                )
+            embeds = embeds + self.pos_embedding[:, : embeds.size(1)]
```

Comment:
here I don't understand how this indexing works with various resolutions (i.e. when we try to generate resolutions other than the fixed sizes corresponding to `sample_height` and `sample_weight`); I think it would work with different number of frames 

because the pos_embedding here contains a 3d positional embedding flatten ( i.e. `frames * height * width`), e.g. if it's generated for 2 frames and size 2 x2, it will have 8 positions, if on the fly we generate 1 frame for size 2x2, we will be able to index correctly since the first 4 positions belong to the first frame; but if we want to generate 2 frames for size 1x2 each, we will also index the first 4 positions, but it would not be correct, no?

Reply:
also here it says it does not support other resolution https://huggingface.co/THUDM/CogVideoX-5b

Reply:
> here I don't...

Yeah... you're right, and tbh I didn't give much thought to it since I tried different frames/resolution first and it seemed to work okay-ish (although 2B is much worse than 5B and understandably so since RoPE is generated on-the-fly). I gave on-the-fly a try and it indeed does produce better results when using different resolutions! Without it, using different resolution and different num_frames produces much weirder patterned videos. Would you like me to push those changes?

> also here it says it does not support other resolution https://huggingface.co/THUDM/CogVideoX-5b

It does say that other resolution/num_frames is not supported but actually seems to work quite well for, say, 57/65 frames from some testing. In regard to the model being trained with only 49 or lower frames, it makes sense to not remove the 49-frame limit. I'm okay with having the limit in case we want to stick with the general guidelines the Cog authors mentioned.

Reply:
> I'm okay with having the limit in case we want to stick with the general guidelines the Cog authors mentioned.

yes let's do this :)